clean_build.sh

```diff
@@ -41,23 +38,15 @@ fi
 # allow local customizations to build environment
 [ -r ./build_env.sh ] && . ./build_env.sh
 
+set -e
+
 # Initialise Git submodules
 git submodule update --init --recursive
 
-rm -rf build
-mkdir build || exit 1
-cd build || exit 1
-echo "Starting Input Leap $B_BUILD_TYPE build..."
-"$B_CMAKE" $B_CMAKE_FLAGS .. || exit 1
-
-if [ "$CMAKE_HAS_PARALLEL" -eq "1" ]; then
```

Comment:
This complication here is to support building on older distributions. E.g. Ubuntu 18.04 only has cmake 3.10, whereas `--parallel` and friends became supported since 3.12.

Reply:
yep, but [the documentation](https://cmake.org/cmake/help/latest/manual/cmake.1.html#build-tool-mode) points at `CMAKE_BUILD_PARALLEL_LEVEL` being the override when `--parallel` is not given:

> The [CMAKE_BUILD_PARALLEL_LEVEL](https://cmake.org/cmake/help/latest/envvar/CMAKE_BUILD_PARALLEL_LEVEL.html#envvar:CMAKE_BUILD_PARALLEL_LEVEL) environment variable, if set, specifies a default parallel level when this option is not given.

iow, we no longer need the check for versions because old cmake won't care about some random env var and new cmake handles it correctly.

Reply:
Indeed, your solution is equivalent to the previous code in less lines. I missed that `ninja` would run using all cores anyway.

Reply:
> I missed that ninja would run using all cores anyway.

so, interestingly enough: we're now actually *limiting* the number of cores to be used. I picked 32 as arbitrary max to be nice on beefy runners (on gitlab.fdo we have `FDO_CI_CONCURRENT` that specifies how many you can use in a job).

If we just want to run with whatever is the default, we can drop this code altogether because then ninja will just use whatever its built-in default is. We could export `MAKEFLAGS=-j` to ensure make does run in parallel where we fall back to that. but for ninja, no work is needed.

Reply:
That's good point, we don't need unnecessary limiting. Myself I thought the reason you wanted that is to have a limit on some CI. Given that 32 threads is plenty for such a small project, I didn't think much about this limitation. But if there's no need for it after all, let's remove it.

Reply:
Done in #1502